prototype_experiment.py

```diff
@@ -0,0 +1,279 @@
+import io
+
+import matplotlib.image as mpimg
+import matplotlib.pyplot as plt
+import nn_lib.models.utils
+import torch
+import torch.nn as nn
+
+
+
+import nn_lib.models.graph_module_plus as gmp
+import nn_lib.models.fancy_layers as fl
+
+from torchvision.transforms import Resize
+from nn_lib.models.utils import conv2d_shape_inverse
+#from nn_lib.analysis.similarity import Conv1x1StitchingLayer, create_stitching_model
+from nn_lib.datasets import ImageNetDataModule, CocoDetectionDataModule
+from nn_lib.models import get_pretrained_model, graph_utils
+from torch.fx import symbolic_trace, GraphModule, graph
+from torch.nn import Conv2d, functional
+from tqdm.auto import tqdm
+
+
+#from stitching_demo import quick_run_and_check, display_model_graph
+
+import seaborn #for beatiful graphs
+import pandas as pd
+
+#Knob #1
+#independent variable is the dataset
+imagenet = ImageNetDataModule(root_dir="/data/datasets", batch_size = 100, num_workers = 10)
+coco_seg = CocoDetectionDataModule(root_dir="/data/datasets", batch_size = 100, num_workers = 10)
+datalist = {"imagenet": imagenet, "coco": coco_seg}
+
+#Knob #1.5
+#The selection of the stitching parents from the pretrained models for the given dataset
+imagenet_models = {"resnet50": get_pretrained_model("resnet18"), "resnet101": get_pretrained_model("resnet34"), "ViT": get_pretrained_model("vit_b_32")}
+coco_models = {}
+modellist = {"imagenet": imagenet_models, "coco": coco_models}
+
+#Knob 2
+#The selection of the stitching layer
+stitching_families = {"conv1x1": Conv2d}
+
+#Knob 3 
+#The selection of the loss function
+loss_functions = {"class_labels": 0, "soft_labels": 1}
```

Comment:
Not a big deal, but a more standard pattern would be to specify these various options in command line arguments like

    parser.add_argument("--loss", type=str, default="class_labels", choices=["class_labels", "soft_labels"])